<?xml version="1.0" encoding="utf-8"?>
<robot name="dummy_psm" xmlns:xacro="http://www.ros.org/wiki/xacro">
	<link name="l_0"/>
	<link name="l_1"/>
	<link name="l_2"/>
	<link name="l_3"/>
	<link name="l_4"/>
	<link name="l_5"/>
	<link name="l_6"/>
	<link name="l_7"/>
	
	<!-- joint 1 -->
	<joint name="outer_yaw" type="revolute">
		<parent link="l_0"/>
		<child link="l_1"/>
		<axis xyz="0 0 1"/>
		<origin rpy="0 -1.5708 1.5708" xyz="0.0 0.0 0.0"/>
		<dynamics damping="1"/> 
		<limit lower="-1.5707" upper="1.5707" velocity=".4" effort="1000"/>
	</joint>
	<!-- joint 2: outer_pitch -->
	<joint name="outer_pitch" type="revolute">
		<parent link="l_1"/>
		<child link="l_2"/>
		<axis xyz="0 0 1"/>
		<origin rpy="-1.5708 -1.5708 0" xyz="0 0 0"/>
		<dynamics damping="1"/>
		<limit lower="-0.7854" upper="0.7854" velocity=".4" effort="1000"/>
	</joint>
	<!-- joint 3: insertion -->
	<joint name="outer_insertion" type="prismatic">
		<parent link="l_2"/>
		<child link="l_3"/>
		<axis xyz="0 0 1"/>
		<origin rpy="1.5708 0 0" xyz="0 0.4318 0"/>
		<limit lower="0" upper="0.240" velocity=".4" effort="1000"/>
		<dynamics damping="1"/> 
	</joint>
	<!-- joint 4: Outer Roll -->
	<joint name="outer_roll" type="revolute">
		<parent link="l_3"/>
		<child link="l_4"/>
		<axis xyz="0 0 1"/>
		<origin rpy="0 0 0" xyz="0 0 0.4162"/>
		<limit lower="-3.0485" upper="3.0485" velocity=".4" effort="1000"/>
		<dynamics damping="1"/>
	</joint>
	<!-- joint 5: Outer wrist pitch joint -->
	<joint name="outer_wrist_pitch" type="revolute">
		<parent link="l_4"/>
		<child link="l_5"/>
		<axis xyz="0 0 1"/>
		<origin rpy="-1.5708 -1.5708 0" xyz="0 0 0"/>
		<limit lower="-1.5707" upper="1.5707" velocity=".4" effort="1000"/>
	</joint>
	<!-- joint 6: Outer wrist yaw joint -->
	<joint name="outer_wrist_yaw" type="revolute">
		<parent link="l_5"/>
		<child link="l_6"/>
		<axis xyz="0 0 1"/>
		<origin rpy="-1.5708 -1.5708 0" xyz="0.0091 0 0"/>
		<!-- +/- 80 deg -->
		<limit lower="-1.3963" upper="1.3963" velocity=".4" effort="1000"/>
	</joint>
	<!-- joint 7-1: Outer open angle 1  -->	
	<joint name="jaw" type="revolute">
		<parent link="l_6"/>
		<child link="l_7"/>
		<axis xyz="0 0 1"/>
		<origin rpy="0 0 0" xyz="0 0 0"/>
		<limit lower="0" upper="1.5707" velocity=".4" effort="1000"/>
	</joint>
</robot>	
